Small fix to emitter of 'data' event

dpovshed opened the pull request:
I believe the $response parameter should be added to not miss very first chunk of data.

Proposed change makes code to work in the same way as the following line 
$response->emit('data', array($bodyChunk, $response));
in the Response::handleData function.

clue commented:
LGTM :+1: 

> I believe the $response parameter should be added to not miss very first chunk of data.
>
> Proposed change makes code to work in the same way as the following line
$response->emit('data', array($bodyChunk, $response));
in the Response::handleData function.

I'm tempted to slightly reword this: We never actually documented the event arguments - this PR makes sure we *consistently* emit the `Response` object as second parameter of its `data` event :)

We might want to update our documentation? https://github.com/reactphp/http-client/blob/9f5e634351ad7f82431afa0568798ad99ec20034/src/Response.php#L12 

`@event($bodyChunk, Response $thisResponse)`

WyriHaximus commented:
Looks mostly good to be me but love to see update to the docs are suggested by @clue and in the [`README.md`](https://github.com/reactphp/http-client/blob/master/README.md#basic-usage) :+1:

dpovshed commented:
Thanks guys for the rapid feedback!
This is my first patch proposed to the phpreact library, so I may be no aware about all necessary procedures. I think I can handle the required fixes to: tests, docs, README.

Or someone else is already on a way to adjust remaining stuff?

WyriHaximus commented:
Congratulations on your first PR 🎉.

Docs and tests are the usual suspects when submitting a PR, that way they
reflect the current state of the code and no one gets confused using it. If
you need help just shout 👍.

dpovshed commented:
Guys, I hope I took into account all propositions, so now PR includes:
- the original fix;
- updated tests;
- doxygen update proposed by @clue ;
- updated example in README.txt (tested - it is working);
- minor typo and punctuation fixes in README.txt

waiting for your feedback! :dart: ;)

WyriHaximus commented:
LGTM :+1: , the type hint is nice to have but not a blocker for me. When @clue and @cboden don't spot any more things I'll merge and tag it.

clue commented:
Good job @dpovshed, LGTM :+1:

dpovshed commented:
Thanks for the review, :) waiting for confirmation and action from @cboden !

cboden commented:
LGTM

WyriHaximus commented:
Thank you for your work, I'll tag it shortly :+1:

WyriHaximus commented:
Tagged [0.4.2](https://github.com/reactphp/http-client/releases/tag/v0.4.2). Thanks again for you work.

dpovshed commented:
Thanks guys for your support! Special to @WyriHaximus for nice blog posts here http://blog.wyrihaximus.net/2015/01/reactphp-introduction/
